frontend/app_flowy/lib/workspace/presentation/settings/widgets/settings_settings_view.dart

```diff
@@ -0,0 +1,48 @@
+import 'package:app_flowy/startup/startup.dart';
+import 'package:flutter/material.dart';
+import 'package:app_flowy/workspace/application/menu/menu_user_bloc.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flowy_sdk/protobuf/flowy-user/protobuf.dart' show UserProfile;
+
+class SettingsSettingsView extends StatelessWidget {
+  final UserProfile user;
+  SettingsSettingsView(this.user, {Key? key}) : super(key: ValueKey(user.id));
+
+  @override
+  Widget build(BuildContext context) {
+    return BlocProvider<MenuUserBloc>(
+      create: (context) => getIt<MenuUserBloc>(param1: user)..add(const MenuUserEvent.initial()),
+      child: BlocBuilder<MenuUserBloc, MenuUserState>(
+        builder: (context, state) => SingleChildScrollView(
+          child: Column(
+            crossAxisAlignment: CrossAxisAlignment.start,
+            children: const [UserNameInput()],
+          ),
+        ),
+      ),
+    );
+  }
+}
+
+class UserNameInput extends StatefulWidget {
```

Comment:
I think putting the underscore in UserNameInput (_UserNameInput) to make it private would be better